parsing/ast_invariants.ml

```diff
@@ -170,6 +170,10 @@ let iterator =
           "In object types, attaching attributes to inherited \
            subtypes is not allowed."
   in
+  let attribute self attr =
+    super.attribute self attr;
+    Builtin_attributes.register_attr ~from_ppx:true attr.attr_name
```

Comment:
This `~from_ppx:true` surprises me. Why is it always from a ppx here?

Reply:
Nice catch.  My initial read of the code was that these invariant checkers are only run on code produced by a ppx.  Looking closer at `pparse.ml`, I see I missed one other case: they are also run if we're parsing an "ast_file" (I think this is just the output of `-dparsetree`).  An obvious solution is to just add a parameter indicating whether we're running these on the output of a ppx or not, but I'd like to check to see if we're ever using ast files for anything first (if so, perhaps there should be three versions of the error instead of two).